hbase-client/src/main/java/org/apache/hadoop/hbase/executor/EventType.java

```diff
@@ -149,7 +149,7 @@
    * C_M_MERGE_REGION<br>
    * Client asking Master to merge regions.
    */
-  C_M_MERGE_REGION          (30, ExecutorType.MASTER_TABLE_OPERATIONS),
+  C_M_MERGE_REGION          (30, ExecutorType.MASTER_MERGE_OPERATIONS),
```

Comment:
One question that came to mind was if upping the size of the TABLE_OPERATIONS pool would have been one way to alleviate your massive merge blockage? (Rather than give it its own pool as here?)

Reply:
Poking around, I see there is specialization for open, close, snapshot, so you are following a pattern.

Reply:
@saintstack It looks like upping the thread-pool backed by MASTER_TABLE_OPERATIONS has other implications, please see the following comment in HMaster. Seems critical that we only have one one thread there, haven't dug deep into it (the comment is decade old, not sure if it still applies).

```  
    // We depend on there being only one instance of this executor running
    // at a time. To do concurrency, would need fencing of enable/disable of
    // tables.
    // Any time changing this maxThreads to > 1, pls see the comment at
    // AccessController#postCompletedCreateTableAction
    this.executorService.startExecutorService(ExecutorType.MASTER_TABLE_OPERATIONS, 1);
```